draft-stark-expect-ct.md

```diff
@@ -23,6 +23,316 @@ TODO: Wite an abstract here
 
 # Introduction
 
+This document defines a new HTTP header that enables UAs to identify web hosts
+that wish to require the presence of Signed Certificate Timestamps (SCTs) in
+future Transport Layer Security (TLS) {{!RFC5246}} connections.
+
+Web hosts that serve the Expect-CT HTTP header are noted by the UA as Expect-CT
+hosts. The UA evaluates each connection to an Expect-CT host for compliance with
+the UA's Certificate Transparency (CT) policy. If the connection violates the CT
+policy, the UA sends a report to a URI configured by the Expect-CT host and/or
+fails the connection, depending on the configuration that the Expect-CT host has
+chosen.
+
+Deploying safely with reporting and gradually increasing max-age. Risk of UAs
+changing policies, servers should use reporting to discover.
+
+Meant to be used with HSTS but they can be used separately.
+
+Expect-CT is a trust-on-first-use (TOFU) mechanism etc etc
+
+## Requirements Language
+
+# Server and Client Behavior
+
+## Response Header Field Syntax
+
+The "Expect-CT" header field is a new response header defined in this
+specification. It is used by a server to indicate that UAs should evaluate
+connections to the host emitting the header for CT compliance
+({{expect-ct-compliance}}).
+
+(I don't know how to do fancy ABNF figures.)
+
+~~~
+Expect-CT-Directives = directive *( OWS ";" OWS directive )
+directive            = directive-name [ "=" directive-value ]
+directive-name       = token
+directive-value      = token / quoted-string
+~~~
+
+The directives defined in this specification are described below. The overall
+requirements for directives are:
+
+1. The order of appearance of directives is not significant.
+
+2. A given directive MUST NOT appear more than once in a given header
+   field. Directives are either optional or required, as stipulated in their
+   definitions.
+
+3.  Directive names are case insensitive.
+
+4.  UAs MUST ignore any header fields containing directives, or other header
+    field value data, that do not conform to the syntax defined in this
+    specification.  In particular, UAs must not attempt to fix malformed header
+    fields.
+
+5.  If a header field contains any directive(s) the UA does not recognize, the
+    UA MUST ignore those directives.
+
+6.  If the Expect-CT header field otherwise satisfies the above requirements (1
+    through 5), the UA MUST process the directives it recognizes.
+
+### The report-uri Directive
+
+The OPTIONAL `report-uri` directive indicates the URI to which the UA SHOULD
+report Expect-CT failures ({{expect-ct-compliance}}). The UA POSTs the reports
+to the given URI as described in {{reporting-expect-ct-failure}}.
+
+Hosts may set `report-uri`s that use HTTP or HTTPS. If the scheme in the
+`report-uri` is one that uses TLS (e.g., HTTPS), UAs MUST check Expect-CT
+compliance when the host in the `report-uri` is an Expect-CT host; similarly,
+UAs MUST apply HSTS if the host in the `report-uri` is a Known HSTS Host.
```

Comment:
We may get in trouble about making a normative requirement on HSTS in this document.